### PR TITLE
[DOC] Refactor change logs

### PR DESCRIPTION
**Pull request by NicolasGensollen**
This PR is a proposal for a refactoring of the change logs, largely inspired by what MNE folks are doing for their own logs, which results in [very nice and handy logs](https://mne.tools/dev/whats_new.html#version-0-24-0-2021-11-03).

More precisely, this PR tries to address the following problems with the current logs:

- The current logs are entirely included in `doc/whats_new.rst` resulting in a very large file. 
- Formatting have changed over the years and is not consistent over releases.
- Old releases have no sphinx label and can't be referenced.
- A lot of entries do not link to the functions or classes they changed.
- It is hard and time consuming to link an entry of the log to the corresponding PR. Doing a git blame of `whats_new.rst` can help a little bit, but it can be a painful experience when someone else has changed the entry since the initial author (fixing a typo for example).
- Contributors used to have their names in a list at the end of (some) releases, sometimes in alphabetical order, sometimes sorted by number of contributions. Having the name of the PR author for each entry of the log makes it more clear who contributed what and when.

The proposal consists in:

- Splitting `whats_new.rst`: one file per release.
- Enforcing a consistent structure for each release.
- Linking to objects from the API whenever possible.
- Adding one file linking contributor profiles which can be included whenever contributor names need to be linked (for example in `AUTHORS.rst`).
- Having each log entry reference the corresponding PR and author.

This is still WIP but comments and suggestions are welcome to improve it.

### CODE REVIEW

File: AUTHORS.rst [hunk 1]
```diff
@@ -20,20 +22,20 @@ Core developers
...
+* `Elizabeth DuPre`_
+* `Gael Varoquaux`_
+* `Hao-Ting Wang`_
+* `Jerome Dockes`_
+* `Julia Huntenburg`_
+* `Kamalakar Reddy Daddy`_
+* `Kshitij Chawla`_
+* `Nicolas Gensollen`_
+* `Taylor Salo`_
+* `Thomas Bazeille`_
```
Comment (bthirion) [inline AUTHORS.rst:39]: Kamalaker and Kshitij is now former members of that group.

File: AUTHORS.rst [hunk 2]
```diff
@@ -69,8 +71,8 @@ Some other past or present contributors are:
 Funding
 .......
 
-Alexandre Abraham, Gael Varoquaux, Kamalakar Reddy Daddy, Loïc Estève,
-Mehdi Rahim, Philippe Gervais were paid by the `NiConnect
+`Alexandre Abraham`_, `Gael Varoquaux`_, `Kamalakar Reddy Daddy`_, `Loic Estève`_,
+`Mehdi Rahim`_, `Philippe Gervais`_ were paid by the `NiConnect
 <https://team.inria.fr/parietal/18-2/spatial_patterns/niconnect>`_
```
Comment (bthirion) [inline AUTHORS.rst:76]: Here we'll have to add the support by HBP for you.
+ the support of Inria for Kshitij.

File: doc/sphinxext/gh_substitutions.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+from docutils.nodes import reference
+from docutils.parsers.rst.roles import set_classes
+
+
+# adapted from
+# https://github.com/mne-tools/mne-python/blob/main/doc/sphinxext/gh_substitutions.py
+
+
+def gh_role(name, rawtext, text, lineno, inliner, options={}, content=[]):
+    """Link to a GitHub issue."""
```
Comment (bthirion) [inline doc/sphinxext/gh_substitutions.py:12]: Either call it `_gh_role` or write complete docstrings

**Review by bthirion — COMMENTED** [review]
A good PR, thx !

File: doc/sphinxext/gh_substitutions.py (hunk 3 above)
Comment (NicolasGensollen) [reply doc/sphinxext/gh_substitutions.py:12]: Thanks! Fixed in 1d6749e

File: AUTHORS.rst (hunk 2 above)
Comment (NicolasGensollen) [reply AUTHORS.rst:76]: I added these two fundings in 3c51c01
Let me know if you think I should change the wording.
Also, it looks like the link to NiConnect is dead: https://team.inria.fr/parietal/18-2/spatial_patterns/niconnect

**Review by bthirion — COMMENTED** [review]
LGTM otherwise

**Review by bthirion — APPROVED** [review]
I didn't spot any outstanding issue. I'll take another look once the doc is generated

**Comment by NicolasGensollen** [comment]
Thanks! Merging....

**Pull request merged**